File: backend/src/main/java/com/paperai/model/entity/FlowDefinition.java
package com.paperai.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 流程定义实体
 *
 * @author ch
 * @date 2026年05月14日
 */
@Data
@TableName("flow_definition")
public class FlowDefinition implements java.io.Serializable {
    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 所属用户 */
    private Long userId;

    /** 流程名称 */
    private String name;

    /** 流程描述 */
    private String description;

    /** 分类：preset/custom/template */
    private String category;

    /** 完整节点 + 边 JSON */
    private String graphData;

    /** 是否为模板 */
    private Integer isTemplate;

    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /** 更新时间 */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: backend-py/app/models/flow_definition.py
from sqlalchemy import String, Text, Integer, BigInteger, JSON, Index
from sqlalchemy.orm import Mapped, mapped_column
from .base import Base, TimestampMixin


class FlowDefinition(Base, TimestampMixin):
    __tablename__ = "flow_definition"
    __table_args__ = (
        Index("ix_fd_user_id", "user_id"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    description: Mapped[str | None] = mapped_column(String(500))
    category: Mapped[str] = mapped_column(String(50), nullable=False, default="custom")
    graph_data: Mapped[dict] = mapped_column(JSON, nullable=False)
    is_template: Mapped[int] = mapped_column(Integer, default=0)
